frontend/src/components/dashboard/FreeOnboarding.tsx

```diff
@@ -0,0 +1,481 @@
+        <Image src={Plus} alt="" />
+      </Button>
+    );
+  }
+
+  if (props.profile.onboarding_state === 1) {
+    const skipMeskTesting = () => {
+      props.onNextStep(3);
+    };
+
+    const nextStep = () => {
+      props.onNextStep(2);
+      gaEvent({
+        category: "Free Onboarding",
+        action: "Engage",
+        label: "onboarding-step-2-next",
+        value: 2,
+      });
+    };
+
+    const forwardedEmail = () => {
+      props.onNextStep(2);
+      gaEvent({
+        category: "Free Onboarding",
+        action: "Engage",
+        label: "onboarding-step-2-continue",
+        value: 2,
+      });
+    };
+
+    step = (
+      <StepTwo
+        aliases={props.aliases}
+        profile={props.profile}
+        user={props.user}
+        runtimeData={props.runtimeData}
+        continue={forwardedEmail}
+        isModalOpen={isModalOpen}
+        setIsModalOpen={setIsModalOpen}
+      />
+    );
+
+    next = (
+      <button
+        ref={nextStepTwoButtonRef}
+        className={styles["next-link"]}
+        onClick={nextStep}
+      >
+        {l10n.getString("profile-free-onboarding--next-step")}
+        <ChevronRightIcon className={styles.chevron} width={16} alt="" />
+      </button>
+    );
+
+    button = (
+      <Button
+        className={styles["generate-new-mask"]}
+        onClick={() => {
+          setIsModalOpen(true);
+        }}
+      >
+        {l10n.getString(
+          "profile-free-onboarding--copy-mask-how-forwarding-works",
+        )}
+        <Image src={Plus} alt="" />
+      </Button>
+    );
+
+    skipButton = (
+      <button
+        ref={skipStepTwoButtonRef}
+        className={styles["skip-link"]}
+        onClick={skipMeskTesting}
+      >
+        {l10n.getString("profile-free-onboarding--skip-step")}
+      </button>
+    );
+  }
+
+  if (props.profile.onboarding_state === 2) {
+    const linkForBrowser = supportsChromeExtension()
+      ? "https://chrome.google.com/webstore/detail/firefox-relay/lknpoadjjkjcmjhbjpcljdednccbldeb?utm_source=fx-relay&utm_medium=onboarding&utm_campaign=install-addon"
+      : "https://addons.mozilla.org/en-CA/firefox/addon/private-relay/";
+
+    const skipAddonStep = () => {
+      props.onNextStep(3);
+    };
+
+    const finish = () => {
+      props.onNextStep(3);
+      gaEvent({
+        category: "Free Onboarding",
+        action: "Engage",
+        label: "onboarding-step-3-complete",
+        value: 3,
+      });
+    };
+
+    step = <StepThree />;
+
+    next = (
+      <button
+        ref={nextStepTwoButtonRef}
+        className={styles["next-link"]}
+        onClick={finish}
+      >
+        {l10n.getString("profile-free-onboarding--addon-finish")}
+        <ChevronRightIcon className={styles.chevron} width={16} alt="" />
+      </button>
+    );
+
+    button = (
+      <>
+        <LinkButton
+          href={linkForBrowser}
+          target="_blank"
```

Comment:
[From what I've read](https://stackoverflow.com/questions/50709625/link-with-target-blank-and-rel-noopener-noreferrer-still-vulnerable), major browsers now automatically use the behavior of `rel="noopener"` and for any `target="_blank"`. And that should cover it since `noreferrer` includes the functionality of `noopener`.